docs/reference/api/connectivity/drivers/EMACMemoryManager.md

```diff
@@ -0,0 +1,31 @@
+## EMAC memory manager
+
+The Ethernet MAC memory manager class provides abstracted memory interface toward memory modules used in different network stacks. The EMAC device driver uses the memory interface to handle memory related operations, such as allocation, free and data manipulation.
+
+Memory buffer chains store the data on the memory interface. The data passed in either direction either may be contiguous (a single-buffer chain) or may consist of multiple buffers. You can chain the buffers using a singly linked list.
+
+### Usage
+
+On link output to EMAC driver, buffer chain ownership is given to the driver. Driver must free the buffer after it is no longer used. The driver can express alignment preferences for buffers sent to it, but the network stack is not required to meet these preferences. This means that the drivers relying on alignment, may need to copy the output data into an aligned (or contiguous) buffers.
+
+EMAC driver can allocate memory either from a memory pool or from the heap (contiguous memory). By preference, link input memory should be allocated from the pool, but if contiguous memory is required it can be allocated from the heap. 
```

Comment:
Query: Who or what should allocate link input memory from the pool? You, the user? Also, who or what might require contiguous memory? If contiguous memory's necessary, who or what can allocate link input memory from the heap?